### PR TITLE
Spelling

### PR DESCRIPTION
**Pull request by jsoref**
Generated by https://github.com/jsoref/spelling `f`; to maintain your repo, please consider `fchurn`

### CODE REVIEW

File: cmd/helm/flags.go [hunk 1]
```diff
@@ -31,7 +31,7 @@ import (
 const outputFlag = "output"
 
 func addValueOptionsFlags(f *pflag.FlagSet, v *values.Options) {
-	f.StringSliceVarP(&v.ValueFiles, "values", "f", []string{}, "specify values in a YAML file or a URL(can specify multiple)")
+	f.StringSliceVarP(&v.ValuesFiles, "values", "f", []string{}, "specify values in a YAML file or a URL(can specify multiple)")
```
Comment (jsoref) [inline cmd/helm/flags.go]: This is more consistent with `valuesfile(s)` elsewhere, along with `values` in this line.

File: cmd/helm/root_test.go [hunk 2]
```diff
@@ -31,7 +31,7 @@ func TestRootCmd(t *testing.T) {
 
 	tests := []struct {
 		name, args, cachePath, configPath, dataPath string
-		envars                                      map[string]string
+		envvars                                     map[string]string
```
Comment (jsoref) [inline cmd/helm/root_test.go]: `envvars` is more common

File: pkg/chartutil/chartfile.go [hunk 3]
```diff
@@ -43,7 +43,7 @@ func LoadChartfile(filename string) (*chart.Metadata, error) {
 // 'filename' should be the complete path and filename ('foo/Chart.yaml')
 func SaveChartfile(filename string, cf *chart.Metadata) error {
 	// Pull out the dependencies of a v1 Chart, since there's no way
-	// to tell the serialiser to skip a field for just this use case
+	// to tell the serializer to skip a field for just this use case
```
Comment (jsoref) [inline pkg/chartutil/chartfile.go]: This is more common in this repository.

File: pkg/cli/values/options.go [hunk 4]
```diff
@@ -30,7 +30,7 @@ import (
 )
 
 type Options struct {
-	ValueFiles   []string
+	ValuesFiles  []string
```
Comment (bacongobbler) [inline pkg/cli/values/options.go]: This is a breaking change to a public API. Can you please revert this?

File: pkg/chartutil/doc.go [hunk 5]
```diff
@@ -17,15 +17,15 @@ limitations under the License.
 /*Package chartutil contains tools for working with charts.
 
 Charts are described in the protocol buffer definition (pkg/proto/charts).
-This packe provides utilities for serializing and deserializing charts.
+This package provides utilities for serializing and deserializing charts.
```
Comment (jsoref) [inline pkg/chartutil/doc.go]: I'm guessing based on line 17

File: pkg/chartutil/errors_test.go [hunk 6]
```diff
@@ -20,15 +20,15 @@ import (
 	"testing"
 )
 
-func TestErrorNoTableDoesntPanic(t *testing.T) {
+func TestErrorNoTableDoesNotPanic(t *testing.T) {
```
Comment (jsoref) [inline pkg/chartutil/errors_test.go]: I'm willing to drop any commit if people object.

File: pkg/storage/driver/driver.go [hunk 7]
```diff
@@ -39,44 +39,44 @@ type Creator interface {
 	Create(key string, rls *rspb.Release) error
 }
 
-// Updator is the interface that wraps the Update method.
+// Updater is the interface that wraps the Update method.
 //
 // Update updates an existing release or returns
 // ErrReleaseNotFound if the release does not exist.
-type Updator interface {
+type Updater interface {
```
Comment (bacongobbler) [inline pkg/storage/driver/driver.go]: This is a breaking change to a public API. Can you please revert this?

File: pkg/storage/driver/driver.go [hunk 8]
```diff
@@ -39,44 +39,44 @@ type Creator interface {
...
 	Update(key string, rls *rspb.Release) error
 }
 
-// Deletor is the interface that wraps the Delete method.
+// Deleter is the interface that wraps the Delete method.
 //
 // Delete deletes the release named by key or returns
 // ErrReleaseNotFound if the release does not exist.
-type Deletor interface {
+type Deleter interface {
```
Comment (bacongobbler) [inline pkg/storage/driver/driver.go]: This is a breaking change to a public API. Can you please revert this?

File: pkg/storage/driver/driver.go [hunk 9]
```diff
@@ -39,44 +39,44 @@ type Creator interface {
...
 	Delete(key string) (*rspb.Release, error)
 }
 
-// Queryor is the interface that wraps the Get and List methods.
+// Querier is the interface that wraps the Get and List methods.
```
Comment (bacongobbler) [inline pkg/storage/driver/driver.go]: This is a breaking change to a public API. Can you please revert this?

File: pkg/chartutil/jsonschema.go [hunk 10]
```diff
@@ -39,10 +39,10 @@ func ValidateAgainstSchema(chrt *chart.Chart, values map[string]interface{}) err
 		}
 	}
 
-	// For each dependency, recurively call this function with the coalesced values
-	for _, subchrt := range chrt.Dependencies() {
-		subchrtValues := values[subchrt.Name()].(map[string]interface{})
-		if err := ValidateAgainstSchema(subchrt, subchrtValues); err != nil {
+	// For each dependency, recursively call this function with the coalesced values
+	for _, subchart := range chrt.Dependencies() {
```
Comment (jsoref) [inline pkg/chartutil/jsonschema.go]: This spelling is used more frequently.

File: pkg/storage/driver/driver.go [hunk 11]
```diff
@@ -39,44 +39,44 @@ type Creator interface {
...
 //
 // Get returns the release named by key or returns ErrReleaseNotFound
 // if the release does not exist.
 //
 // List returns the set of all releases that satisfy the filter predicate.
 //
 // Query returns the set of all releases that match the provided label set.
-type Queryor interface {
+type Querier interface {
```
Comment (bacongobbler) [inline pkg/storage/driver/driver.go]: This is a breaking change to a public API. Can you please revert this?

**Review by bacongobbler — CHANGES_REQUESTED** [review]

File: pkg/chartutil/testdata/subpop/charts/subchart1/values.yaml [hunk 12]
```diff
@@ -23,7 +23,7 @@ overridden-chartA:
   SCAbool: true
   SCAfloat: 3.14
   SCAint: 100
-  SCAstring: "jabathehut"
+  SCAstring: "jabbathehut"
```
Comment (jsoref) [inline pkg/chartutil/testdata/subpop/charts/subchart1/values.yaml]: I'm assuming this is a typo and not something intentional.

I generally hope that places which intentionally misspell things include very clear language indicating why they're misspelling things.

File: pkg/plugin/installer/http_installer_test.go [hunk 13]
```diff
@@ -234,12 +234,12 @@ func TestExtract(t *testing.T) {
 	gz.Close()
 	// END tarball creation
 
-	extr, err := NewExtractor(source)
+	extractor, err := NewExtractor(source)
```
Comment (jsoref) [inline pkg/plugin/installer/http_installer_test.go]: The tested code uses this spelling...

File: pkg/repo/doc.go [hunk 14]
```diff
@@ -27,7 +27,7 @@ The first is the 'index.yaml' format, which is expressed like this:
 	entries:
 	  frobnitz:
 	  - created: 2016-09-29T12:14:34.830161306-06:00
-		description: This is a frobniz.
+		description: This is a frobnitz.
```
Comment (jsoref) [inline pkg/repo/doc.go]: This is the actual name of the module...

File: pkg/repo/index_test.go [hunk 15]
```diff
@@ -131,7 +131,7 @@ func TestMerge(t *testing.T) {
 
 	if len(ind1.Entries) != 2 {
 		t.Errorf("Expected 2 entries, got %d", len(ind1.Entries))
-		vs := ind1.Entries["dreadnaught"]
+		vs := ind1.Entries["dreadnought"]
```
Comment (jsoref) [inline pkg/repo/index_test.go]: afaict, this code is unreachable/untested/would not work.

See lines 116/122.

I could be wrong, but as w/ `jabbathehut`, if there's a specific reason for misspelling something, it should be clearly identified.

File: pkg/storage/driver/driver.go [hunk 16]
```diff
@@ -39,44 +39,44 @@ type Creator interface {
 	Create(key string, rls *rspb.Release) error
 }
 
-// Updator is the interface that wraps the Update method.
+// Updater is the interface that wraps the Update method.
```
Comment (jsoref) [inline pkg/storage/driver/driver.go]: I'm happy to drop such things, however, ... https://wikidiff.com/updater/updator

File: pkg/storage/driver/driver.go (hunk 9 above)
Comment (jsoref) [inline pkg/storage/driver/driver.go]: I half expect to get the most push-back (in this hunk) on this correction, but...

File: pkg/storage/driver/driver.go (hunk 11 above)
Comment (jsoref) [reply pkg/storage/driver/driver.go]: Sure

Comment (jsoref) [reply pkg/storage/driver/driver.go]: I presume this applies to the entire family (Updat*/Delet*/Quer*)

**Comment by jsoref** [comment]
@bacongobbler: I think I've dropped the objectionable commits.

**Comment by jsoref** [comment]
(sorry, mouse missed... -- not a fan of how easy MSGitHub makes it for one to accidentally do that)

**Review by bacongobbler — APPROVED** [review]

**Review by hickeyma — APPROVED** [review]
LGTM, thanks @jsoref

**Pull request merged**